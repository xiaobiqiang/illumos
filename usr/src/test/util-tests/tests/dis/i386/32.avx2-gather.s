/*
 * This file and its contents are supplied under the terms of the
 * Common Development and Distribution License ("CDDL"), version 1.0.
 * You may only use this file in accordance with the terms of version
 * 1.0 of the CDDL.
 *
 * A full copy of the text of the CDDL should have accompanied this
 * source.  A copy of the CDDL is also available via the Internet at
 * http://www.illumos.org/license/CDDL.
 */

/*
 * Copyright 2016 Joyent, Inc.
 */

/*
 * Test AVX2 Gather related instructions
 */

.text
.align 16
.globl libdis_test
.type libdis_test, @function
libdis_test:
	vgatherdpd	%xmm0, (%eax, %xmm1, 4), %xmm2
	vgatherdpd	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vgatherdpd	%ymm0, (%eax, %xmm1, 4), %ymm2
	vgatherdpd	%ymm0, 0x42(, %xmm1, 4), %ymm2
	vgatherdps	%xmm0, (%eax, %xmm1, 4), %xmm2
	vgatherdps	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vgatherdps	%ymm0, (%eax, %ymm1, 4), %ymm2
	vgatherdps	%ymm0, 0x42(, %ymm1, 4), %ymm2
	vgatherqpd	%xmm0, (%eax, %xmm1, 4), %xmm2
	vgatherqpd	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vgatherqpd	%ymm0, (%eax, %ymm1, 4), %ymm2
	vgatherqpd	%ymm0, 0x42(, %ymm1, 4), %ymm2
	vgatherqps	%xmm0, (%eax, %xmm1, 4), %xmm2
	vgatherqps	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vgatherqps	%xmm0, (%eax, %ymm1, 4), %xmm2
	vgatherqps	%xmm0, 0x42(, %ymm1, 4), %xmm2
	vpgatherdd	%xmm0, (%eax, %xmm1, 4), %xmm2
	vpgatherdd	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vpgatherdd	%ymm0, (%eax, %ymm1, 4), %ymm2
	vpgatherdd	%ymm0, 0x42(, %ymm1, 4), %ymm2
	vpgatherdq	%xmm0, (%eax, %xmm1, 4), %xmm2
	vpgatherdq	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vpgatherdq	%ymm0, (%eax, %xmm1, 4), %ymm2
	vpgatherdq	%ymm0, 0x42(, %xmm1, 4), %ymm2
	vpgatherqd	%xmm0, (%eax, %xmm1, 4), %xmm2
	vpgatherqd	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vpgatherqd	%xmm0, (%eax, %ymm1, 4), %xmm2
	vpgatherqd	%xmm0, 0x42(, %ymm1, 4), %xmm2
	vpgatherqq	%xmm0, (%eax, %xmm1, 4), %xmm2
	vpgatherqq	%xmm0, 0x42(, %xmm1, 4), %xmm2
	vpgatherqq	%ymm0, (%eax, %ymm1, 4), %ymm2
	vpgatherqq	%ymm0, 0x42(, %ymm1, 4), %ymm2
.size libdis_test, [.-libdis_test]
