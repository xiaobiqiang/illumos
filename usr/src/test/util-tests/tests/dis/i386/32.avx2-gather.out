    libdis_test:       c4 e2 f9 92 14 88  vgatherdpd %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0x6:   c4 e2 f9 92 14 8d  vgatherdpd %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x10:  c4 e2 fd 92 14 88  vgatherdpd %ymm0,(%eax,%xmm1,4),%ymm2
    libdis_test+0x16:  c4 e2 fd 92 14 8d  vgatherdpd %ymm0,0x42(,%xmm1,4),%ymm2
                       42 00 00 00 
    libdis_test+0x20:  c4 e2 79 92 14 88  vgatherdps %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0x26:  c4 e2 79 92 14 8d  vgatherdps %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x30:  c4 e2 7d 92 14 88  vgatherdps %ymm0,(%eax,%ymm1,4),%ymm2
    libdis_test+0x36:  c4 e2 7d 92 14 8d  vgatherdps %ymm0,0x42(,%ymm1,4),%ymm2
                       42 00 00 00 
    libdis_test+0x40:  c4 e2 f9 93 14 88  vgatherqpd %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0x46:  c4 e2 f9 93 14 8d  vgatherqpd %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x50:  c4 e2 fd 93 14 88  vgatherqpd %ymm0,(%eax,%ymm1,4),%ymm2
    libdis_test+0x56:  c4 e2 fd 93 14 8d  vgatherqpd %ymm0,0x42(,%ymm1,4),%ymm2
                       42 00 00 00 
    libdis_test+0x60:  c4 e2 79 93 14 88  vgatherqps %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0x66:  c4 e2 79 93 14 8d  vgatherqps %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x70:  c4 e2 7d 93 14 88  vgatherqps %xmm0,(%eax,%ymm1,4),%xmm2
    libdis_test+0x76:  c4 e2 7d 93 14 8d  vgatherqps %xmm0,0x42(,%ymm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x80:  c4 e2 79 90 14 88  vpgatherdd %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0x86:  c4 e2 79 90 14 8d  vpgatherdd %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0x90:  c4 e2 7d 90 14 88  vpgatherdd %ymm0,(%eax,%ymm1,4),%ymm2
    libdis_test+0x96:  c4 e2 7d 90 14 8d  vpgatherdd %ymm0,0x42(,%ymm1,4),%ymm2
                       42 00 00 00 
    libdis_test+0xa0:  c4 e2 f9 90 14 88  vpgatherdq %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0xa6:  c4 e2 f9 90 14 8d  vpgatherdq %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0xb0:  c4 e2 fd 90 14 88  vpgatherdq %ymm0,(%eax,%xmm1,4),%ymm2
    libdis_test+0xb6:  c4 e2 fd 90 14 8d  vpgatherdq %ymm0,0x42(,%xmm1,4),%ymm2
                       42 00 00 00 
    libdis_test+0xc0:  c4 e2 79 91 14 88  vpgatherqd %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0xc6:  c4 e2 79 91 14 8d  vpgatherqd %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0xd0:  c4 e2 7d 91 14 88  vpgatherqd %xmm0,(%eax,%ymm1,4),%xmm2
    libdis_test+0xd6:  c4 e2 7d 91 14 8d  vpgatherqd %xmm0,0x42(,%ymm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0xe0:  c4 e2 f9 91 14 88  vpgatherqq %xmm0,(%eax,%xmm1,4),%xmm2
    libdis_test+0xe6:  c4 e2 f9 91 14 8d  vpgatherqq %xmm0,0x42(,%xmm1,4),%xmm2
                       42 00 00 00 
    libdis_test+0xf0:  c4 e2 fd 91 14 88  vpgatherqq %ymm0,(%eax,%ymm1,4),%ymm2
    libdis_test+0xf6:  c4 e2 fd 91 14 8d  vpgatherqq %ymm0,0x42(,%ymm1,4),%ymm2
                       42 00 00 00 
